### PR TITLE
tests: Add Kubernetes test to check UPC and IPC namespaces.

### LINKED ISSUE #332: Add a k8s test to check that containers in a pod share UTS and IPC namespaces
This could be a crio or k8s test. We need to launch two or more containers in a pod, and check that the uts and IPC namespaces for the two containers are the same.
Basically run `ls -la /proc/1/ns/uts` and `ls -la /proc/1/ns/ipc` in both the containers and make sure that the namespace identifier is the same.
(Make sure that pid namespaces are not shared in the configuration, which is the case by default).

First, https://github.com/kata-containers/agent/pull/249 needs to be merged.

### PR DESCRIPTION
**Pull request by GabyCT**
This verifies that the UPC and IPC namespaces are the same.

Fixes #332

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/kubernetes/busybox.bats [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	first_pod_status_cmd="sudo -E kubectl get pods -a | grep $first_pod_name | grep Running"
+	waitForProcess "$wait_time" "$sleep_time" "$first_pod_status_cmd"
+	first_complete_pod_name=$(sudo -E kubectl get pods | grep "$first_pod_name" | cut -d ' ' -f1)
+	# Run the second pod
+	sudo -E kubectl run $second_pod_name --image=$busybox_image -- sh  -c "eval $sleep_cmd"
+	second_pod_status_cmd="sudo -E kubectl get pods -a | grep $second_pod_name | grep Running"
+	waitForProcess "$wait_time" "$sleep_time" "$second_pod_status_cmd"
+	second_complete_pod_name=$(sudo -E kubectl get pods | grep "$second_pod_name" | cut -d ' ' -f1)
+	# Check UTS namespace
+	uts_cmd="ls -la /proc/1/ns/uts"
```
Comment (jodh-intel) [inline integration/kubernetes/busybox.bats]: Why aren't these commands using `/proc/self/ns/...`?

File: integration/kubernetes/busybox.bats [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	# Pull the images before launching workload. This is mainly because we use
+	# a timeout and in slow networks it may result in not been able to pull the image
+	# successfully.
+	sudo -E crictl pull "$busybox_image"
+}
+
+@test "Check UTS and IPC namespaces" {
+	wait_time=30
+	sleep_time=5
+	# Run the first pod
```
Comment (jodh-intel) [inline integration/kubernetes/busybox.bats]: I find this very hard to read as it's all in a single block - could you maybe add some blank lines to break the block into logical chunks/paragraphs?

File: integration/kubernetes/run_kubernetes_tests.sh [hunk 3]
```diff
@@ -21,5 +21,6 @@ fi
 pushd "$kubernetes_dir"
 ./init.sh
 bats nginx.bats
+bats busybox.bats
```
Comment (jodh-intel) [inline integration/kubernetes/run_kubernetes_tests.sh]: The name of this file doesn't give any indication as to what the test is doing. I'd suggest renaming it to something like `k8s-uts+ipc-ns.bats`.

File: integration/kubernetes/k8s-uts+ipc-ns.bats [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (jodh-intel) [inline integration/kubernetes/k8s-uts+ipc-ns.bats]: Nit: I'd remove this as strictly it isn't a bash script any more.

**Comment by devimc** [comment]
CI is not happy

```
1..1
not ok 1 Check UTS and IPC namespaces
# (from function `waitForProcess' in file ../../.ci/lib.sh, line 153,
#  in test file k8s-uts+ipc-ns.bats, line 29)
#   `waitForProcess "$wait_time" "$sleep_time" "$first_pod_status_cmd"' failed
# Image is update to date for docker.io/library/busybox@sha256:74f634b1bc1bd74535d5209589734efbd44a25f4e2dc96d78784576a3eb5b335
# deployment.apps "first-test" created
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Flag --show-all has been deprecated, will be removed in an upcoming release
# Error from server (NotFound): pods "first-test" not found
# Error from server (NotFound): pods "second-test" not found
# NAME                             READY     STATUS        RESTARTS   AGE
# first-test-6c9475dc95-9s6ww      1/1       Running       0          32s
# nginx-service-75ccfbd5c9-bwkvn   0/1       Terminating   0          1m
```

File: integration/kubernetes/k8s-uts+ipc-ns.bats (hunk 4 above)
Comment (chavafg) [reply integration/kubernetes/k8s-uts+ipc-ns.bats]: maybe change it to: #!/usr/bin/env bats

File: integration/kubernetes/k8s-uts+ipc-ns.bats [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+	# Check IPC namespace
+	ipc_cmd="ls -la /proc/self/ns/ipc"
+	first_pod_ipc_namespace=$(sudo -E kubectl exec "$first_complete_pod_name" -- sh -c "$ipc_cmd" | grep ipc | cut -d ':' -f3)
+	second_pod_ipc_namespace=$(sudo -E kubectl exec "$second_complete_pod_name" -- sh -c "$ipc_cmd" | grep ipc | cut -d ':' -f3)
+	[ "$first_pod_ipc_namespace" == "$second_pod_ipc_namespace" ]
+}
+
+teardown() {
+	sudo -E kubectl delete pod "$first_pod_name"
```
Comment (chavafg) [inline integration/kubernetes/k8s-uts+ipc-ns.bats]: change `delete pod` to `delete deployment`, so the pod is not restarted, same in the line below.

**Comment by chavafg** [comment]
lgtm

**Pull request merged**